### PR TITLE
Provide initial support policy for KEDA

### PR DESCRIPTION
**Pull request by tomkerkhove**
Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

Provide initial support policy for KEDA to help people understand how to get help, FAQ, troubleshoot and our Kubernetes support model.

Once this is merged, a PR will be opened to link to it on keda.sh.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [ ] Tests have been added
- [ ] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [ ] Changelog has been updated

### CODE REVIEW

File: SUPPORT.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## Slack
+
+* [Slack](https://kubernetes.slack.com) ([registration](http://slack.k8s.io)):
+The `#keda` channel is usually the place where people offer support.
+
+## Kubernetes Support
+
+The supported window of Kubernetes versions with KEDA is known as "N-2": (N (Latest release) - 2 (minor versions)).
+
+However, maintainers can dedide to extend this by supporting more minor versions based on the required CRDs being used; but there is no guarantee.
```
Comment (zroubalik) [inline SUPPORT.md:20]:
```suggestion
However, maintainers can decide to extend this by supporting more minor versions based on the required CRDs being used; but there is no guarantee.
```

File: SUPPORT.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Support for deploying and using KEDA
+
+Welcome to KEDA! We use GitHub for tracking bugs and feature requests but here are some pointers to get help.
```
Comment (zroubalik) [inline SUPPORT.md:3]: We should be more vocal, that the support is "best effort only", it is not a commercial support or something like this, that could be expected by someone.

File: SUPPORT.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+* [FAQ](https://keda.sh/docs/latest/faq/)
+
+## Slack
+
+* [Slack](https://kubernetes.slack.com) ([registration](http://slack.k8s.io)):
+The `#keda` channel is usually the place where people offer support.
+
+## Kubernetes Support
+
+The supported window of Kubernetes versions with KEDA is known as "N-2": (N (Latest release) - 2 (minor versions)).
```
Comment (zroubalik) [inline SUPPORT.md:18]: I am not sure if this is true, or if we even can say something like this. Currently KEDA can run on k8s >= 1.16 and I don't see us bringing any new features, that will break support on that version.

And we don't actively test against Kubernetes version (N-2), we run our tests on 1.18 only.

File: SUPPORT.md (hunk 2 above)
Comment (tomkerkhove) [reply SUPPORT.md:3]: Agreed, I will.

File: SUPPORT.md (hunk 3 above)
Comment (tomkerkhove) [reply SUPPORT.md:18]: The reason why I've added it here is mainly for us having to comply with at least N-2 support and not use new features that are available in version N. This helps customers understand what to expect and relates to https://github.com/kedacore/charts/issues/104 for example.

Makes sense or not?

Comment (zroubalik) [reply SUPPORT.md:18]: OKay :)

File: SUPPORT.md (hunk 2 above)
Comment (tomkerkhove) [reply SUPPORT.md]: PTAL

File: SUPPORT.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+* [Troubleshooting Guide](https://keda.sh/docs/2.0/troubleshooting/)
+* [FAQ](https://keda.sh/docs/latest/faq/)
+
+## Community
+
+We have a nice community that is always happy to help each other:
+
+* [Slack](https://kubernetes.slack.com) ([registration](http://slack.k8s.io)):
+The `#keda` channel is usually the place where people offer support.
+* [Stack Overflow](https://stackoverflow.com/questions/tagged/keda)
```
Comment (zroubalik) [inline SUPPORT.md:21]: Shouldn't we link GitHub discussions in here?

**Review by zroubalik — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

**Pull request merged**